universal-application-tool-0.0.1/conf/application.conf

```diff
@@ -306,9 +306,7 @@ play.evolutions {
   db.default {
     useLocks = true
     autoApply = true
-    # This might break something in the prod database someday.  Let's remove it once we completely agree
-    # never to require any more evolution downs.
-    autoApplyDowns = true
+    autoApplyDowns = false
```

Comment:
Question (non-blocking): in what scenario would downs be applied when this is set to true? Could it happen when a server starts or does it only happen when the servers are shutdown?

Reply:
Read the doc. I see. the evolutions are applied when the scripts are changed.

Reply:
I *think* some of the current ups are not idempotent though, like the ones that add a column.

Reply:
Hm, that's a good point - what's the equivalent to "if not exists" for adding a column.

Reply:
(It's not a big deal since we use evolution locks, but it would be nice to know!)

Reply:
I *think* you can do

```sql
ALTER TABLE table_name ADD [COLUMN] IF NOT EXISTS column_name data_type;
```

Reply:
Ah, good eye!  I would do it to the old evolutions but as of this PR, we can't do that anymore, ha.